### PR TITLE
Add deployment steps in test CI

### PR DESCRIPTION
**Pull request by CahidArda**

### CODE REVIEW

**Comment by CahidArda** [comment]
mismatch between the nextjs version of the sdk and the user causes this error from [CI](https://github.com/upstash/qstash-js/actions/runs/11184281140/job/31094689797?pr=195)

> Property '[INTERNALS]' is missing in type 'import("/home/runner/work/qstash-js/qstash-js/node_modules/next/dist/server/web/spec-extension/request").NextRequest' but required in type 'import("/home/runner/work/qstash-js/qstash-js/examples/nextjs/node_modules/next/dist/server/web/spec-extension/request").NextRequest'.

[reference](https://docs.arcjet.com/troubleshooting#property-internals-is-missing-in-type)

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -99,16 +161,110 @@ jobs:
...
+          version: latest
+
+      - name: Deploy
+        run: |
+          pnpm add @upstash/qstash@${{needs.release.outputs.version}}
+          DEPLOYMENT_URL=$(npx vercel --token=${{ secrets.VERCEL_TOKEN }})
+          echo "DEPLOYMENT_URL=${DEPLOYMENT_URL}" >> $GITHUB_ENV
+        env:
+          VERCEL_ORG_ID: ${{secrets.VERCEL_TEAM_ID}}
+          VERCEL_PROJECT_ID: "prj_dUPhtwNlX6otLH5CSV4N0CsZzoaq"
```
Comment (fahreddinozcan) [inline .github/workflows/test.yaml:215]: can we move `VERCEL_PROJECT_ID` to action secrets? We don't need to expose it here, also would be easier to change later on.

Comment (CahidArda) [reply .github/workflows/test.yaml:215]: I think we can, but then we should do it in all CI PRs, and open one for Redis too

Shall we do it another time? :D

Comment (ytkimirti) [reply .github/workflows/test.yaml:215]: Yeah I thought of that too :D 
Maybe when we need to change the id we can fix it :)

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -52,27 +52,89 @@ jobs:
...
+        run: bun test ci.test.ts
+        working-directory: examples/cloudflare-workers
+        env:
+          DEPLOYMENT_URL: https://upstash-qstash.upsdev.workers.dev
+
   nextjs-local-build:
     runs-on: ubuntu-latest
+    # disabled because of a local build issue. Will be tested properly in
+    # the deployment test
+    if: false
```
Comment (ytkimirti) [inline .github/workflows/test.yaml:137]: What is the local build issue?

**Review by ytkimirti — APPROVED** [review]

File: .github/workflows/test.yaml (hunk 2 above)
Comment (CahidArda) [reply .github/workflows/test.yaml:137]: similar to https://github.com/upstash/ratelimit-js/pull/123#discussion_r1789718986

it works fine in release, but local installation has a type error. I don't want to write @ts-expect-error in order to not miss it in release, so I disabled the local one until we figure this out

**Pull request merged**